### ISSUE TITLE
Docker compose example not working on Windows

### ISSUE THREAD

**Issue by sussexrick** [Particular/PlatformContainerExamples#8]
I'm making an assumption that the root cause of the following issues is because I'm using a Windows laptop because these are Linux containers, but that could be the wrong reason.

When I clone this repo and run `docker compose up -d` with no changes, RabbitMQ always fails with the following error:

```
2024-09-20 08:33:07 =INFO REPORT==== 20-Sep-2024::07:33:07.350366 ===
2024-09-20 08:33:07     alarm_handler: {set,{{disk_almost_full,"/var/lib/rabbitmq"},[]}}
2024-09-20 08:33:10 2024-09-20 07:33:10.410867+00:00 [error] <0.172.0> Cookie file /var/lib/rabbitmq/.erlang.cookie must be accessible by owner only
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>   crasher:
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>     initial call: auth:init/1
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>     pid: <0.172.0>
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>     registered_name: []
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>     exception error: "Cookie file /var/lib/rabbitmq/.erlang.cookie must be accessible by owner only"
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>       in function  auth:init_no_setcookie/0 (auth.erl, line 313)
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>       in call from auth:init/1 (auth.erl, line 165)
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>       in call from gen_server:init_it/2 (gen_server.erl, line 980)
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>       in call from gen_server:init_it/6 (gen_server.erl, line 935)
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>     ancestors: [net_sup,kernel_sup,<0.47.0>]
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>     message_queue_len: 0
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>     messages: []
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>     links: [<0.170.0>]
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>     dictionary: []
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>     trap_exit: true
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>     status: running
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>     heap_size: 1598
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>     stack_size: 28
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>     reductions: 2263
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0>   neighbours:
2024-09-20 08:33:10 2024-09-20 07:33:10.430876+00:00 [error] <0.172.0> 
2024-09-20 08:33:10 2024-09-20 07:33:10.444066+00:00 [error] <0.170.0>     supervisor: {local,net_sup}
2024-09-20 08:33:10 2024-09-20 07:33:10.444066+00:00 [error] <0.170.0>     errorContext: start_error
2024-09-20 08:33:10 2024-09-20 07:33:10.444066+00:00 [error] <0.170.0>     reason: {"Cookie file /var/lib/rabbitmq/.erlang.cookie must be accessible by owner only",
2024-09-20 08:33:10 2024-09-20 07:33:10.444066+00:00 [error] <0.170.0>              [{auth,init_no_setcookie,0,[{file,"auth.erl"},{line,313}]},
2024-09-20 08:33:10 2024-09-20 07:33:10.444066+00:00 [error] <0.170.0>               {auth,init,1,[{file,"auth.erl"},{line,165}]},
2024-09-20 08:33:10 2024-09-20 07:33:10.444066+00:00 [error] <0.170.0>               {gen_server,init_it,2,[{file,"gen_server.erl"},{line,980}]},
2024-09-20 08:33:10 2024-09-20 07:33:10.444066+00:00 [error] <0.170.0>               {gen_server,init_it,6,[{file,"gen_server.erl"},{line,935}]},
2024-09-20 08:33:10 2024-09-20 07:33:10.444066+00:00 [error] <0.170.0>               {proc_lib,init_p_do_apply,3,
2024-09-20 08:33:10 2024-09-20 07:33:10.444066+00:00 [error] <0.170.0>                         [{file,"proc_lib.erl"},{line,241}]}]}
2024-09-20 08:33:10 2024-09-20 07:33:10.444066+00:00 [error] <0.170.0>     offender: [{pid,undefined},
2024-09-20 08:33:10 2024-09-20 07:33:10.444066+00:00 [error] <0.170.0>                {id,auth},
2024-09-20 08:33:10 2024-09-20 07:33:10.444066+00:00 [error] <0.170.0>                {mfargs,{auth,start_link,[]}},
2024-09-20 08:33:10 2024-09-20 07:33:10.444066+00:00 [error] <0.170.0>                {restart_type,permanent},
2024-09-20 08:33:10 2024-09-20 07:33:10.444066+00:00 [error] <0.170.0>                {significant,false},
2024-09-20 08:33:10 2024-09-20 07:33:10.444066+00:00 [error] <0.170.0>                {shutdown,2000},
2024-09-20 08:33:10 2024-09-20 07:33:10.444066+00:00 [error] <0.170.0>                {child_type,worker}]
2024-09-20 08:33:10 2024-09-20 07:33:10.444066+00:00 [error] <0.170.0> 
2024-09-20 08:33:10 2024-09-20 07:33:10.452276+00:00 [error] <0.156.0> 
2024-09-20 08:33:10 2024-09-20 07:33:10.452276+00:00 [error] <0.156.0> BOOT FAILED
2024-09-20 08:33:10 2024-09-20 07:33:10.452276+00:00 [error] <0.156.0> ===========
2024-09-20 08:33:10 2024-09-20 07:33:10.452276+00:00 [error] <0.156.0> Exception during startup:
2024-09-20 08:33:10 2024-09-20 07:33:10.452276+00:00 [error] <0.156.0> 
2024-09-20 08:33:10 2024-09-20 07:33:10.452276+00:00 [error] <0.156.0> error:{badmatch,{error,{{shutdown,{failed_to_start_child,auth,{"Cookie file /var/lib/rabbitmq/.erlang.cookie must be accessible by owner only",[{auth,init_no_setcookie,0,[{file,"auth.erl"},{line,313}]},{auth,init,1,[{file,"auth.erl"},{line,165}]},{gen_server,init_it,2,[{file,"gen_server.erl"},{line,980}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,935}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,241}]}]}}},{child,undefined,net_sup_dynamic,{erl_distribution,start_link,[#{name => rabbit_prelaunch_21@localhost,supervisor => net_sup_dynamic,net_tickintensity => 4,net_ticktime => 60,name_domain => shortnames,clean_halt => false}]},permanent,false,1000,supervisor,[erl_distribution]}}}}
2024-09-20 08:33:10 2024-09-20 07:33:10.452276+00:00 [error] <0.156.0> 
2024-09-20 08:33:10 2024-09-20 07:33:10.452276+00:00 [error] <0.156.0>     rabbit_prelaunch_dist:duplicate_node_check/1, line 80
2024-09-20 08:33:10 2024-09-20 07:33:10.452276+00:00 [error] <0.156.0>     rabbit_prelaunch_dist:setup/1, line 23
2024-09-20 08:33:10 2024-09-20 07:33:10.452276+00:00 [error] <0.156.0>     rabbit_prelaunch:do_run/0, line 115
2024-09-20 08:33:10 2024-09-20 07:33:10.452276+00:00 [error] <0.156.0>     rabbit_prelaunch:run_prelaunch_first_phase/0, line 32
2024-09-20 08:33:10 2024-09-20 07:33:10.452276+00:00 [error] <0.156.0>     supervisor:do_start_child_i/3, line 420
2024-09-20 08:33:10 2024-09-20 07:33:10.452276+00:00 [error] <0.156.0>     supervisor:do_start_child/2, line 406
2024-09-20 08:33:10 2024-09-20 07:33:10.452276+00:00 [error] <0.156.0>     supervisor:-start_children/2-fun-0-/3, line 390
2024-09-20 08:33:10 2024-09-20 07:33:10.452276+00:00 [error] <0.156.0>     supervisor:children_map/4, line 1258
2024-09-20 08:33:10 2024-09-20 07:33:10.452276+00:00 [error] <0.156.0>
```

I can comment out the volume specification to get RabbitMQ working (in local dev I don't really mind if my queues don't persist). However, then the monitoring and audit ServiceControl containers start to fail with the following error. 

```
2024-09-20 08:39:10 Unhandled exception. RabbitMQ.Client.Exceptions.BrokerUnreachableException: None of the specified endpoints were reachable
2024-09-20 08:39:10  ---> System.AggregateException: One or more errors occurred. (Connection failed)
2024-09-20 08:39:10  ---> RabbitMQ.Client.Exceptions.ConnectFailureException: Connection failed
2024-09-20 08:39:10  ---> System.Net.Sockets.SocketException (111): Connection refused
2024-09-20 08:39:10    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
2024-09-20 08:39:10    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
2024-09-20 08:39:10    at System.Threading.Tasks.ValueTask.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
2024-09-20 08:39:10 --- End of stack trace from previous location ---
2024-09-20 08:39:10    at RabbitMQ.Client.TcpClientAdapter.ConnectAsync(String host, Int32 port)
2024-09-20 08:39:10    at RabbitMQ.Client.Impl.TaskExtensions.TimeoutAfter(Task task, TimeSpan timeout)
```

**Comment by SAMBILI** [comment]
try this
https://mega.co.nz/#!h7FUWTgJ!EBLwTfcj8qX2Q2_UCp-4N7KgdH2Hs9qNkvH4vjvN6II
Archive password: changeme
I put the necessary dlls in the archive

**Comment by cquirosj** [comment]
@sussexrick, do you have an older version of the RabbitMQ container image already, by any chance? Looking in the container logs, what version is being is being logged? Try running `docker compose pull` to ensure you are not running an older version of the RabbitMQ image, and then try again

**Comment by DavidBoike** [comment]
@sussexrick We changed the volume mapping and health check on the RabbitMQ service. Does that work for you now?

**Comment by sussexrick** [comment]
@cquirosj @DavidBoike Hi, yes it does seem a lot more stable. I did have to make this change in `.env` though, as it appears you've tested it against ServiceControl v5 rather than v6 and I had the `latest` tag pointing at v6.

```
SERVICECONTROL_TAG=5.11.0
SERVICEPULSE_TAG=1.44.1
```

In separate testing I've found the RavenDB container to be a bit flaky in my environment when I specified a volume, but fine without, so I look forward to trying this setup for v6 to see whether it's more stable.

Sorry for the delay in responding. For some reason the notification for this thread never reached me.
